sqlobject/src/main/java/org/jdbi/v3/sqlobject/DefaultObjectBinder.java

```diff
@@ -19,20 +19,36 @@
 
 class DefaultObjectBinder implements Binder<Bind, Object>
 {
+    private final int paramIndex;
+
+    DefaultObjectBinder()
+    {
+        this(-1);
+    }
+
+    DefaultObjectBinder(int paramIndex)
+    {
+        this.paramIndex = paramIndex;
+    }
+
     @Override
     public void bind(SQLStatement<?> q, Parameter param, Bind b, Object arg)
     {
-        String value = b.value();
+        String value = b != null ? b.value() : Bind.USE_PARAM_NAME;
 
         if (value.equals(Bind.USE_PARAM_NAME)) {
-            if (!param.isNamePresent()) {
+            if (param.isNamePresent()) {
+                value = param.getName();
+            } else if (b == null && paramIndex >= 0) {
+                value = Integer.toString(paramIndex);
+            } else {
                 throw new UnsupportedOperationException("A parameter was annotated with @Bind "
                         + "but no name was specified, and parameter name data is not present "
                         + "in the class file.  " + param.getDeclaringExecutable() + " :: " + param);
             }
-            value = param.getName();
         }
 
+        q.bind(paramIndex, arg);
```

Comment:
The more I look at this, the more I think we don't even want this class doing anything positional.